src/components/PopupWithForm.js

```diff
@@ -16,6 +18,14 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  setIsLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```